### DISCUSSION TITLE
How to use include only on specific lines of a file?

### DISCUSSION THREAD

**Discussion by skwde** [quarto-dev/quarto-cli#10476] [Q&A]
### Description

Sometimes I want to include only certain lines of a file.
E.g. I have a python file with some function definitions.

I want to have only a single function definition in a certain code block.

How do I do that?

I use 1.5.54 on debian bookwork in a devcontainer on macos.

**Comment by mcanouil** [comment]
Use Python to source or possibly the following extension (or a modification of it):
- https://github.com/quarto-ext/include-code-files

**Reply by skwde** [reply]
Unfortunately this does nothing. Not even printing an error when I specify a wrong file name / path.

**Reply by mcanouil** [reply]
> Unfortunately this does nothing. Not even printing an error when I specify a wrong file name / path.

Unfortunately, we can't help with this.

Could you share a **small self-contained "working" (reproducible)** example to work with, _i.e._, a complete Quarto document or a Git repository? The goal is to make it as easy as possible for us to recreate your problem so that we can fix it: please help us help you! Thanks.

---

You can share a **self-contained "working" (reproducible)** Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).
See <https://quarto.org/bug-reports.html#small-is-beautiful-aim-for-a-single-document-with-10-lines>.

If you have multiple files (and if it is **absolutely required** to have multiple files), please share as a Git repository.

<table>
<tr><th>R</th><th>Python</th></tr>
<tr><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: knitr
---

This is a reproducible Quarto document.

{{< lipsum 1 >}}

```{r}
x <- c(1, 2, 3, 4, 5)
y <- c(1, 4, 9, 16, 25)

plot(x, y)
```

![An image]({{< placeholder 600 400 >}}){#fig-placeholder}

{{< lipsum 1 >}}

The end after @fig-placeholder.
````
`````

</td><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: jupyter
---

This is a reproducible Quarto document.

{{< lipsum 1 >}}

```{python}
import matplotlib.pyplot as plt

x = [1, 2, 3, 4, 5]
y = [1, 4, 9, 16, 25]

plt.plot(x, y)
plt.show()
```

![An image]({{< placeholder 600 400 >}}){#fig-placeholder}

{{< lipsum 1 >}}

The end after @fig-placeholder.
````
`````

</td></tr>
</table>

Additionally and if not already given, please share the output of `quarto check` within a code blocks (*i.e.*, using three backticks ` ```txt `), see <https://quarto.org/bug-reports.html#check>.
